### PR TITLE
Private/mmeeks/moretiles

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by mmeeks** [comment]
WIP patch - but would love feedback. I'm looking at re-working tileprocessed - so we can ignore deltas on the wire since they are rather small B/W wise; I'd like to get >=2 items down the connection if possible

File: wsd/DocumentBroker.cpp [hunk 1]
```diff
@@ -3405,8 +3404,7 @@ void DocumentBroker::sendRequestedTiles(const std::shared_ptr<ClientSession>& se
                     tilesNeedsRendering.push_back(tile);
                     _debugRenderedTileCount++;
                 }
-                if (tileCache().subscribeToTileRendering(tile, session, now))
-                    beingRendered++;
+                tileCache().subscribeToTileRendering(tile, session, now);
```
Comment (caolanm) [inline wsd/DocumentBroker.cpp:3407]: presumably the return value from subscribeToTileRendering can be dropped again after that, as no more users of it

Comment (caolanm) [reply wsd/DocumentBroker.cpp:3407]: Dropping the beingRendered would have, presumably, also gotten me out of the hole I found myself in during ed8f95c1bc5ac9a2e5c7dd2fb17d7a94f9d8b2b5 trying to get the empty-update tests in place.

I wonder should we continue to "subscribeToTileRendering" + "sendTileCombine" in handleTileCombinedRequest before calling "sendRequestedTiles" which also does those things? Is that a duplication of effort?

File: wsd/DocumentBroker.cpp [hunk 2]
```diff
@@ -3340,38 +3340,23 @@ void DocumentBroker::sendRequestedTiles(const std::shared_ptr<ClientSession>& se
...
-              // If we delayed all tiles we don't send any tile (we will when next tileprocessed message arrives)
-              delayedTiles < requestedTiles.size())
+        while (!requestedTiles.empty() &&
+               session->getTilesOnFlyCount() < tilesOnFlyUpperLimit)
         {
             TileDesc& tile = *(requestedTiles.begin());
 
-            // We already sent out two versions of the same tile, let's not send the third one
-            // until we get a tileprocessed message for this specific tile.
-            if (session->countIdenticalTilesOnFly(tile) >= 2)
```
Comment (caolanm) [inline wsd/DocumentBroker.cpp:3366]: oddly enough TileCacheTests::testLimitTileVersionsOnFly apparently tries to test for this but seeing as ci is green the test didn't catch this change

**Review by caolanm — APPROVED** [review]

**Pull request merged**